src/test/java/uk/gov/pay/ledger/performance/PerformanceDaoIT.java

```diff
@@ -0,0 +1,61 @@
+package uk.gov.pay.ledger.performance;
+
+import com.google.gson.Gson;
+import org.hamcrest.Matchers;
+import org.junit.ClassRule;
+import org.junit.Test;
+import uk.gov.pay.ledger.performance.dao.PerformanceDao;
+import uk.gov.pay.ledger.performance.entity.PerformanceReportEntity;
+import uk.gov.pay.ledger.rule.AppWithPostgresAndSqsRule;
+import uk.gov.pay.ledger.transaction.state.TransactionState;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Map;
+
+import static java.math.BigDecimal.ZERO;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.number.BigDecimalCloseTo.closeTo;
+import static uk.gov.pay.ledger.util.fixture.TransactionFixture.aTransactionFixture;
+
+public class PerformanceDaoIT {
+
+    private static Gson gson = new Gson();
+
+    @ClassRule
+    public static AppWithPostgresAndSqsRule rule = new AppWithPostgresAndSqsRule();
+
+    private PerformanceDao transactionDao = new PerformanceDao(rule.getJdbi());
+
+    @Test
+    public void report_volume_total_amount_and_average_amount() {
+        aTransactionFixture().withAmount(1000L).withState(TransactionState.STARTED).withTransactionType("PAYMENT")
+                .withTransactionDetails(toJsonString(Map.of("live", true))).insert(rule.getJdbi());
+
+        aTransactionFixture().withAmount(1000L).withState(TransactionState.SUCCESS).withTransactionType("REFUND")
+                .withTransactionDetails(toJsonString(Map.of("live", true))).insert(rule.getJdbi());
+
+        aTransactionFixture().withAmount(1000L).withState(TransactionState.SUCCESS).withTransactionType("PAYMENT")
+                .withTransactionDetails(toJsonString(Map.of("live", false))).insert(rule.getJdbi());
+
+        List<Long> relevantAmounts = List.of(1200L, 1020L, 750L);
+        relevantAmounts.stream().forEach(amount -> aTransactionFixture()
+                .withAmount(amount)
+                .withState(TransactionState.SUCCESS)
+                .withTransactionType("PAYMENT")
+                .withTransactionDetails(toJsonString(Map.of("live", true)))
+                .insert(rule.getJdbi()));
+        BigDecimal expectedTotalAmount = new BigDecimal(relevantAmounts.stream().mapToLong(amount -> amount).sum());
+        BigDecimal expectedAverageAmount = new BigDecimal(relevantAmounts.stream().mapToLong(amount -> amount).average().getAsDouble());
+
+        PerformanceReportEntity performanceReport = transactionDao.performanceReportForPaymentTransactions();
+        assertThat(performanceReport.getTotalVolume(), is(3L));
+        assertThat(performanceReport.getTotalAmount(), Matchers.is(closeTo(expectedTotalAmount, ZERO)));
+        assertThat(performanceReport.getAverageAmount(), Matchers.is(closeTo(expectedAverageAmount, ZERO)));
+    }
+
+    private static String toJsonString(Map<String, ?> map) {
+        return gson.toJson(map);
+    }
+}
```

Comment:
[minor] Valid test - feels slightly overcomplicated to me.
I prefer explicit assertions (instead of calculating the `expectedAverageAmount` and  `expectedTotalAmount`).
I'd rather add few easy to calculate avg values and set the value explicitly in the assertion.

Reply:
Are there any reasons for the preference of explicit assertions?
I like the way its done here as I can see the expected values are calculated from the `relevantAmounts` list. This way I can easily deduce that the first three transactions that are inserted into the test are irrelevant.

Reply:
test should not (or rarely should) do any calculations - it increases the risk of testing the code with the same code (for example: testing the average function with the assertion that uses exactly the same function to calculate the expected value).
Explicit is easier to read.
Good names are 👍, but explicit assertions are easier to read and verify (imho).